pkg/watcher/results/results.go

```diff
@@ -116,6 +120,31 @@ func (c *Client) ensureResult(ctx context.Context, o Object, opts ...grpc.CallOp
 		}
 	}
 
+	// Set the Result.Annotations and Result.Summary.Annotations fields if
+	// the object in question contains the required annotations.
+
+	if value, found := o.GetAnnotations()[annotation.ResultAnnotations]; found {
+		if annotations, err := parseAnnotations(annotation.ResultAnnotations, value); err != nil {
+			return nil, err
+		} else {
+			if res.Annotations == nil {
+				res.Annotations = make(map[string]string, len(annotations))
+			}
+			copyKeys(annotations, res.Annotations)
+		}
+	}
+
+	if value, found := o.GetAnnotations()[annotation.RecordSummaryAnnotations]; found {
+		if annotations, err := parseAnnotations(annotation.RecordSummaryAnnotations, value); err != nil {
+			return nil, err
+		} else if res.Summary != nil {
+			if res.Summary.Annotations == nil {
```

Comment:
Same issue here. We should first copy `curr.Summary.Annotations` to `res.Summary.Annotations` .